stdlib/public/runtime/Casting.cpp

```diff
@@ -2755,6 +2755,51 @@ swift::swift_conformsToProtocol(const Metadata *type,
   goto recur;
 }
 
+/// Return the type metadata for a given type name.
+///
+/// This is a temporary implementation to support some Foundation use cases;
+/// it will only lookup a type that directly implements a protocol as only
+/// those types will have conformance table entries.
+static const Metadata *
+_metadataForName(const llvm::StringRef &typeName) {
+  installCallbacksToInspectDylib();
```

Comment:
I left installCallbacksToInspectDylib() in both places for now, as it deadlocks when called from the Conformances constructor because the Lazy container is still being initialised.